src/hooks/useLoadDocuments.ts

```diff
@@ -17,23 +17,24 @@ export interface DocumentTypeFromServer {
   alternatives: AlternativeWordTypeFromServer[]
 }
 
-const transformArticle = (documents: ReturnType<DocumentTypeFromServer[]>): ReturnType<DocumentType[]> => {
-  const formattedDocuments: DocumentType[] =
-    documents.data?.map(document => ({
-      ...document,
-      article: ARTICLES[document.article],
-      alternatives: document.alternatives.map(it => ({
-        article: ARTICLES[it.article],
-        word: it.alt_word
-      }))
-    })) ?? []
-  return { ...documents, data: formattedDocuments }
-}
+const formatServerResponse = (documents: DocumentTypeFromServer[]): DocumentType[] =>
+  documents.map(document => ({
+    ...document,
+    article: ARTICLES[document.article],
+    alternatives: document.alternatives.map(it => ({
+      article: ARTICLES[it.article],
+      word: it.alt_word
+    }))
+  })) ?? []
 
-const useLoadDocuments = (discipline: Discipline): ReturnType<DocumentType[]> => {
+export const loadDocuments = async (discipline: Discipline): Promise<DocumentType[]> => {
   const url = ENDPOINTS.documents.replace(':id', `${discipline.id}`)
-  const documents: ReturnType<DocumentTypeFromServer[]> = useLoadFromEndpoint(url, discipline.apiKey)
-  return transformArticle(documents)
+
+  const response = await getFromEndpoint<DocumentTypeFromServer[]>(url, discipline.apiKey)
+  return formatServerResponse(response)
 }
 
+const useLoadDocuments = (discipline: Discipline): ReturnType<DocumentType[]> =>
+  useLoadFromEndpoint(loadDocuments, discipline)
```

Comment:
Is this "PingPong" really needed?
-> In useLoadDocuments call method from useLoadFromEndpoint
-> in useLoadFromEndpoint call method (loadDocuments) from useLoadDocuments
-> In useLoadDocuments (loadDocuments-method) call method (getFromEndpoint) from useLoadFromEndpoint

Would it maybe make more sense to just pass the formatting-method to useLoadFromEndpoint?

Reply:
I think that is not as it is, there should be no function call of `loadDocuments` in `useLoadFromEndpoint` :thinking: So we just have dependencies from `useLoadDocuments` to `useLoadFromEndpoint` (and their other functions), so only one way.

Reply:
yes, got this, but it gets passed to it and is then executed. My point is more about the execution order and not the dependencies.

Reply:
A I see. Yeah not sure about a better way. If we just pass the formatting method it won't be possible to use the fetch logic without hook.